slasher/rpc/detect_update_min_max_span.go

```diff
@@ -0,0 +1,115 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+type detect func(attestationEpochSpan uint64, recorderEpochSpans *ethpb.MinMaxEpochSpan, attestationSourceEpoch uint64) uint64
```

Comment:
Please add more detail on this type and why it is needed, typically this would be a better fit for something like an interface like:
```go
type Detector interface {
  Detect()
}
```
and create a struct that matches that interface. Then, the interface can be an input argument anywhere it is needed, allowing for a nice abstraction over detection itself. This interface pattern is more idiomatic with Go.